components/movementsensor/adxl345/adxl345.go

```diff
@@ -294,7 +508,13 @@ func (adxl *adxl345) Accuracy(ctx context.Context, extra map[string]interface{})
 func (adxl *adxl345) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	adxl.mu.Lock()
 	defer adxl.mu.Unlock()
-	return map[string]interface{}{"linear_acceleration": adxl.linearAcceleration}, adxl.err.Get()
+
+	readings := make(map[string]interface{})
+	readings["linear_acceleration"] = adxl.linearAcceleration
+	readings["single_tap"] = fmt.Sprintf("count: %v, last_seen: %v", adxl.interruptsFound[SingleTap].count, adxl.interruptsFound[SingleTap].lastSeen)
+	readings["freefall"] = fmt.Sprintf("count: %v, last_seen: %v", adxl.interruptsFound[FreeFall].count, adxl.interruptsFound[FreeFall].lastSeen)
+
+	return readings, adxl.err.Get()
```

Comment:
I think I am missing something here. It looks like you should be able to pass custom types from the proto file but am getting an error